src/main/java/gov/cabinetoffice/gap/applybackend/service/SubmissionService.java

```diff
@@ -192,9 +201,14 @@ public boolean isSubmissionReadyToBeSubmitted(final String userId, final UUID su
             return false;
         }
 
-        return submission
+        final List<SubmissionSection> sections = submission
                 .getDefinition()
-                .getSections()
+                .getSections();
+
+        final boolean sectionAreAllCompleted = sections.stream().filter(section -> !section.getSectionStatus().equals(SubmissionSectionStatus.COMPLETED)).toList().isEmpty();
```

Comment:
you could use `stream.filter.anyMatch` here instead. Would also recommend dropping the stream down onto new lines rather than having it on one line like this.